src/nvim/tui/tui.c

```diff
@@ -348,50 +370,210 @@ static void update_attrs(UI *ui, HlAttrs attrs)
       data->params[1].i = (fg >> 8) & 0xff;   // green
       data->params[2].i = fg & 0xff;          // blue
       unibi_out(ui, data->unibi_ext.set_rgb_foreground);
+      data->default_attr = false;
```

Comment:
Instead of repeating this we could set it once:

    data->default_attr = !(attrs.bold || ... || fg == -1 || bg == -1);